public/app/features/alerting/unified/components/rules/RuleActionsButtons.tsx

```diff
@@ -131,6 +136,11 @@ export const RuleActionsButtons = ({ compact, showViewButton, showCopyLinkButton
         }}
       />
       {deleteModal}
+      {isGrafanaRulerRule(rule.rulerRule) && showSilenceDrawer && (
```

Comment:
I believe the In-rule silence drawer should be only available for GMA. Mimir/Loki data sources doesn't support that fine-grained permissions and rule-level silences not to mention issues with linking rules generators to alertmanagers
Having said that, I do wonder if we should display the Alertmanager selector in the drawer at all
Technically GMA alerts can be sent to external AMs but then the userflow gets weirs -> The user should silence for each alertmanager separately
wdyt @gillesdemey @soniaAguilarPeiron

Reply:
RBAC - I almost forgot about it 😄 
RBAC makes it even more tricky - The idea is that the drawer allows to only create so called rule-level silences which are silences tied to a specific rule and cannot impact alerts from other rules
However these rule-level silences are only supported by Grafana Alertmanager so it might be odd to display the drawer for external alertmanagers 🤔

Reply:
I believe we should limit the display of the 'silence alert rule' drawer to scenarios where it's specifically used for silencing a Grafana-managed alert rule (by alert rule UID). There is no alert rule uid for cloud alerts.
However, I see the value in the drawer as an enhancement for creating silences, whether they are from Grafana Alertmanager or not. Therefore, it might be beneficial to expand the drawer's availability to include cloud scenarios as well. In this case, we would adjust the title to 'Create a Silence' and exclude the alert UID matcher functionality.